test/functional/wallet_avoidreuse.py

```diff
@@ -240,5 +242,37 @@ def test_fund_send_fund_send(self):
         assert_approx(self.nodes[1].getbalance(), 1, 0.001)
         assert_approx(self.nodes[1].getbalance(avoid_reuse=False), 11, 0.001)
 
+    def test_many_utxos_one_dest(self):
+        '''
+        Test the case where [1] generates a new address A, then
+        [0] sends 1 BTC to A 30 times.
+        [1] spends 11 BTC from A
+        20 of the outputs in A should be used, leaving 10 intact.
+        Those 10 should all be marked used.
+        A single output of 9 btc should be change back to self, and should not be marked used.
```

Comment:
nits: s/change/changed/, and s/self/itself/ ?

Reply:
It's change, as in money given back, so I think it's correct as it is, but I'll make it clearer.

Reply:
Oh, of course. Yes, I read it wrong.